index.d.ts

```diff
@@ -8,18 +8,22 @@ export interface StoreCreator {
   ): Store<S>;
 }
 
-export type Loop<S, A extends Action> = [S, CmdType<A>];
+export type Loop<S, A extends Action = never> = [S, CmdType<A>];
 
-export interface LoopReducer<S, A extends Action = AnyAction> {
+export interface LoopReducerDeprecated<S, A extends Action = AnyAction> {
```

Comment:
it's probably worth throwing in a comment explaining why. Maybe link to https://github.com/reduxjs/redux/issues/3580 and copy a buggy example from the issue as a comment?

```ts
type MyActionTypes = ActionType1 | ActionType2; //implicitly ActionType1 | ActionType2 | never

const MyReducer: Reducer<MyState, MyActionTypes> = (state = initState, action) => {
  switch(action.type){
    case actionTypes.TYPE_1:
      return handleType1(state, action); //infers that action is ActionType1

    case actionTypes.TYPE_2:
      return handleType2(state, action); //infers that action is ActionType2
    
	default:
	  return action; //infers that action type is never, so it does not complain that the return type is invalid because it assumes this will never happen anyway. you can also just skip the default case entirely and it won't complain
  } 
};
```